File: agent-ui/src/types/llmTrace.ts
export interface LlmTraceSummary {
  id: number;
  trace_id: string;
  session_id: string;
  user_input: string;
  final_output: string | null;
  model_name: string;
  provider: string | null;
  status: string;
  duration_ms: number | null;
  total_tokens: number | null;
  created_at: string;
}

export interface LlmTraceMessage {
  id: number;
  trace_id: string;
  session_id: string;
  message_order: number;
  role: string;
  content: string;
  message_type: string;
  tool_name: string | null;
  created_at: string;
}

export interface LlmTraceDetail {
  conversation: LlmTraceSummary;
  messages: LlmTraceMessage[];
}

File: llm/schemas/llm_trace.py
from pydantic import BaseModel


class LlmTraceSummary(BaseModel):
    id: int
    trace_id: str
    session_id: str
    user_input: str
    final_output: str | None
    model_name: str
    provider: str | None
    status: str
    duration_ms: int | None
    total_tokens: int | None
    created_at: str


class LlmTraceMessage(BaseModel):
    id: int
    trace_id: str
    session_id: str
    message_order: int
    role: str
    content: str
    message_type: str
    tool_name: str | None
    created_at: str


class LlmTraceDetail(BaseModel):
    conversation: LlmTraceSummary
    messages: list[LlmTraceMessage]
